python/tvm/relay/frontend/common.py

```diff
@@ -451,20 +451,27 @@ def get_name(node):
     return name
 
 
-def infer_type(node):
+def infer_type(node, mod=None):
     """A method to infer the type of an intermediate node in the relay graph."""
-    mod = node if isinstance(node, _module.Module) else _module.Module.from_expr(node)
+    if isinstance(mod, _module.Module):
+        mod["main"] = _expr.Function([], node)
```

Comment:
Create a new module here instead of modifying the existing one?

Reply:
improved.